UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -0,0 +1,627 @@
+          DEBUG ((DEBUG_INFO, "  Found ranges Property TempLen (%08X)\n", TempLen));
+
+          PciRootBridgeInfo->RootBridge[index].AllocationAttributes = EFI_PCI_HOST_BRIDGE_COMBINE_MEM_PMEM | EFI_PCI_HOST_BRIDGE_MEM64_DECODE;
+          PciRootBridgeInfo->RootBridge[index].Supports             = ROOT_BRIDGE_SUPPORTS_DEFAULT;
+          PciRootBridgeInfo->RootBridge[index].PMemAbove4G.Base     = PcdGet64 (PcdPciReservedPMemAbove4GBBase);
+          PciRootBridgeInfo->RootBridge[index].PMemAbove4G.Limit    = PcdGet64 (PcdPciReservedPMemAbove4GBLimit);
+          PciRootBridgeInfo->RootBridge[index].PMem.Base            = PcdGet32 (PcdPciReservedPMemBase);
+          PciRootBridgeInfo->RootBridge[index].PMem.Limit           = PcdGet32 (PcdPciReservedPMemLimit);
+          PciRootBridgeInfo->RootBridge[index].UID                  = index;
+          PciRootBridgeInfo->RootBridge[index].HID                  = EISA_PNP_ID (0x0A03);
+
+          Data32                                                = (UINT32 *)(PropertyPtr->Data);
+          PciRootBridgeInfo->RootBridge[index].Mem.Base         = Fdt32ToCpu (*(Data32 + 2));
+          PciRootBridgeInfo->RootBridge[index].Mem.Limit        = Fdt32ToCpu (*(Data32 + 6)) + Fdt32ToCpu (*(Data32 + 2)) -1;
+          PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base  = Fdt32ToCpu (*(Data32 + 9)) + LShiftU64 (Fdt32ToCpu (*(Data32 + 8)), 32);
+          PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit = PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base + LShiftU64 (Fdt32ToCpu (*(Data32 + 12)), 32) +  Fdt32ToCpu (*(Data32 + 13)) -1;
+          PciRootBridgeInfo->RootBridge[index].Io.Base          = Fdt32ToCpu (*(Data32 + 16));
+          PciRootBridgeInfo->RootBridge[index].Io.Limit         = PciRootBridgeInfo->RootBridge[index].Io.Base + Fdt32ToCpu (*(Data32 + 20)) -1;
+
+          DEBUG ((DEBUG_INFO, "RootBridgeCount %x, index :%x\n", RootBridgeCount, index));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.Base %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Mem.limit %x, \n", PciRootBridgeInfo->RootBridge[index].Mem.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->MemAbove4G.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].MemAbove4G.Limit));
+
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.Base %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Base));
+          DEBUG ((DEBUG_INFO, "PciRootBridge->Io.limit %llx, \n", PciRootBridgeInfo->RootBridge[index].Io.Limit));
+          if (index > 0) {
+            index--;
+          }
+        }
+
+        if (AsciiStrCmp (TempStr, "bus-range") == 0) {
+          UINT16  *Data16;
+
+          DEBUG ((DEBUG_INFO, "  Found bus-range Property TempLen (%08X)\n", TempLen));
+
+          if (RootBridgeCount == 1) {
+            index = 0;

```

Comment:
why?

Reply:
for  PciRootBridgeInfo->RootBridge[**index**].AllocationAttribute initialize .
if we have only one root bridge , the index should be the 0 , not 1 , or it cause the system assert .
due to PciRootBridgeInfo->RootBridge[0] isn't initialized .

Reply:
But is that the correct way to handle it? I think we should initialize index = RootBridgeCount; As soon as RB is found --index. This will take care of correctly populating all RBs.

Reply:
pci-rb loop has been updated to resolve index issue